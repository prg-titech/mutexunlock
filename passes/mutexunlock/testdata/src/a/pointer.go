package main

import (
	"sync"
)

var pointerOfMutex = &sync.Mutex{}

func PointerOfMutex() {
	pointerOfMutex.Lock()
} // want "missing unlock"

func MutexAsArg(mu *sync.Mutex) {
	mu.Lock()
} // want "missing unlock"
