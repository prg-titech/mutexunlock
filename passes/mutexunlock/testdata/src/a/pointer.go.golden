package main

import (
	"sync"
)

var pointerOfMutex = &sync.Mutex{}

func PointerOfMutex() {
	pointerOfMutex.Lock()
	pointerOfMutex.Unlock()
} // want "missing unlock"

func MutexAsArg(mu *sync.Mutex) {
	mu.Lock()
	mu.Unlock()
} // want "missing unlock"
